lib/config/models/credentials.json

```diff
@@ -0,0 +1,85 @@
+{
+  "$id": "http://express-gateway.io/models/credentials.json",
+  "type": "object",
+  "definitions": {
+    "credentialBase": {
+      "type": "object",
+      "properties": {
+        "autoGeneratePassword": {
+          "type": "boolean",
+          "default": true
+        },
+        "scopes": {
+          "type": [
+            "string",
+            "array"
+          ],
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "autoGeneratePassword"
+      ]
+    }
+  },
+  "properties": {
+    "basic-auth": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/credentialBase"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "passwordKey": {
+              "type": "string",
+              "default": "password"
+            }
+          },
+          "required": [
+            "passwordKey"
+          ]
+        }
+      ]
+    },
+    "key-auth": {
+      "type": "object",
+      "properties": {
+        "scopes": {
+          "type": [
+            "string",
+            "array"
+          ],
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "jwt": {
+      "type": "object",
```

Comment:
JWT has not scopes verification, right? because there is no common field to store them, so every third party has own way. 

scopes are validated only if JWT is a token type for oauth2, correct?

Reply:
That's generally correct but incomplete. Here's the whole story:

By default, JWT payload is not predictable so, as you stated, we do not know where the scopes could be put.

On the other other hand, you can define a scope on a credential type (just like you would do on `basic-auth` or `key-auth`), so you can check gateway scopes (internal scopes?) on it. 

Therefore the idea would be:

1. You create a `jwt` credential type and you put some scopes on it
2. You configure your system of record (Auth0 or whatever) so that the credential id is put in the `sub` standard field of the JWT
3. The gateway [will recognize](https://github.com/ExpressGateway/express-gateway/blob/master/lib/policies/jwt/jwt.js#L31) the sub, look for the credential and check the scopes on it. 

Again — these are scoped coupled to the credential itself wrote in the gateway — it has nothing to do with token scopes that, again, we can't detect.

This check can be disabled using `checkCredentialExistence` parameter to false.

Does that answer your question? I could expand the glitch demo to show the whole story if necessary.

Reply:
thanks for explanation. 

please bring this topic to one of the next standups

I remember some another vortex @altsang @kevinswiber 
as result there is current implementation where scopes are attached to **credential** not to consumer

I personaly like when actual consumer is assigned with permissions
like IAM is doing that. and when you need different scope you create different app with credential 

and when you need some common set of permission\scopes you create role\group

In any case having both ways is quite confusing. So we probably should reenter this vortex

Reply:
All right. I've setup a reminder on Slack so we recall to discuss the feature.